internal/certificates/certificate_operations.go

```diff
@@ -120,3 +105,102 @@ func GenerateCertificateForResource(ctx context.Context, client *api_client.Sddc
 	}
 	return nil
 }
+
+func ReadCertificate(ctx context.Context, client *vcfclient.VcfClient,
+	domainId, resourceFqdn string) (*models.Certificate, error) {
+	viewCertificatesParams := certificates.NewGetCertificatesByDomainParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+	viewCertificatesParams.ID = domainId
+
+	certificatesResponse, _, err := client.Certificates.GetCertificatesByDomain(viewCertificatesParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get certificates by domain: %w", err)
+	}
+
+	// Check if any certificates are found
+	if certificatesResponse.Payload == nil || len(certificatesResponse.Payload.Elements) == 0 {
+		return nil, fmt.Errorf("no certificates found for domain ID %s", domainId)
+	}
+
+	allCertsForDomain := certificatesResponse.Payload.Elements
+	for _, cert := range allCertsForDomain {
+		if cert.IssuedTo != nil && *cert.IssuedTo == resourceFqdn {
+			return cert, nil
+		}
+	}
+	return nil, nil
```

Comment:
This method shouldn't end with both of its results being nil.
You should return an error if no certificate is found.

Reply:
Updated the code and the test shows the change works well.